### ISSUE TITLE
Save / Load systems

### ISSUE THREAD

**Issue by hankjordan** [hankjordan/bevy_save#10]
It may be useful to have systems that are run whenever a save / load is triggered.
These systems could be set up to remap entities or deal with texture handle changes.
Users would then be able to chain these systems together to define save / load _pipelines_ that would be able to replace many of the cumbersome filters.

**Comment by hankjordan** [comment]
> What should be saved to / from disk and what should be initialized on startup?
> 
> This is an unsolved problem in my book and the area that needs the most attention in `bevy_save` - perhaps there should be some sort of initialization system?
>
> The end goal is to make it so that you can use the simple `World::save()` and `World::load()` interface while also having the ability to deal with entity mapping / handle mapping / asset mapping / etc transparently.

#11 has shown that the stop-gap implementation is insufficient and only really works for toy applications.

It may be necessary to introduce some kind of unique ID for entities, but ideally we would be able to get away with just using _pipelines_.

**Comment by hankjordan** [comment]
Refactor `World::save()` and `World::load()` to accept `Pipeline` instead of `&str`

```rust 
fn save<P: Pipeline>(target: P) -> Result<(), SaveableError> { ... }
```

Implement a "default pipeline" for `&str` and `String` - this would behave like the current defaults

```rust
impl<'a> Pipeline for &'a str { /* Default Pipeline */ }
impl Pipeline for String { /* Default Pipeline */ }
```

Deprecate `AppSaver`, `AppLoader`, `AppBackend`, `AppMappingMode`, `AppDespawnMode`, etc

```rust
// ...
```

The user will now be able to re-use different save / load configurations and be able to save / load different groups of entities / resources within the same app

```rust
// User code

#[derive(Pipeline)]
struct ConfigPipeline;

#[derive(Pipeline)]
struct StatePipeline(String);z

// ...

// Default pipeline
world
  .save("save")
  .expect("Failed to save");

// ...

world
  .save(ConfigPipeline)
  .expect("Failed to save config");
  
// ...

world
  .save(StatePipeline("game0".into())
  .expect("Failed to save state");
```

Pipeline types are just identifiers for system chains
```rust
app
  .add_systems(
    (
      load_configs,
      map_config_entities,
    )
    .in_pipeline(ConfigPipeline)
  )
```

Pipeline systems may be added modularly
```rust
app
  .add_system(load_configs.in_pipeline::<ConfigPipeline>())
  
app
  .add_system(map_config_entities.in_pipeline::<ConfigPipeline>())
```

Take advantage of schedules? Above becomes syntax sugar for
```rust
app
  .add_systems(
    (
      load_configs,
      map_config_entities,
    )
    .in_schedule(OnPipeline(ConfigPipeline))
  )
```

**Comment by hankjordan** [comment]
Closed by [#784994](https://github.com/hankjordan/bevy_save/commit/78499450bb3c1babb65fa041dc12a2cd4912a231)
